src/osd/OSD.cc

```diff
@@ -7458,6 +7467,13 @@ void OSD::_committed_osd_maps(epoch_t first, epoch_t last, MOSDMap *m)
 	do_restart = true;
       }
     }
+    if (osdmap->test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS) !=
+	newmap->test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)) {
+      dout(10) << __func__ << " REQUIRE_LUMINOUS flag changed in " << newmap->get_epoch()
+	       << dendl;
+      clear_pg_stat_queue();
```

Comment:
@liewegas , pg enqueues itself based on `PG::get_osdmap()`, but see https://github.com/ceph/ceph/commit/f71c18abbc4e6650ddbb0a72a25ea71697219e1f#diff-dfb9ddca0a3ee32b266623e8fa489626R2756, i disable the enqueue based on the OSD's latest osdmap.